build-n-publish.sh

```diff
@@ -3,6 +3,9 @@
 pkgName="doppler-webapp"
 pkgVersion=${1:-"v0.0.0-build0"}
 cdnBaseUrl=${2:-"//cdn.fromdoppler.com/$pkgName"}
+pkgVersionQa="qa-$pkgVersion"
+pkgVersionInt="int-$pkgVersion"
+pkgVersionDevelopment="dev-$pkgVersion"
```

Comment:
Why do all the builds in one go? Is it really neccesary?

Reply:
Mmm, I am not sure about the meaning of your question.

If you are talking about always publishing for each Doppler environment, I think that it is fine because allow us to test our build with the backend that we want in each moment. And since it is cheaper, I do not see any issue. Do you have any specific concern?

If you are talking about doing it calling _build-n-publish.sh_ more than a time in Jenkings, for example replacing this:

![image](https://user-images.githubusercontent.com/1157864/53012672-6afcde00-3422-11e9-8e1e-b38343cd5ca2.png)

by something this:

```
sh ./build-n-publish.sh build$BUILD_NUMBER //cdn.fromdoppler.com/doppler-webapp production
sh ./build-n-publish.sh qa-build$BUILD_NUMBER //cdn.fromdoppler.com/doppler-webapp qa
sh ./build-n-publish.sh int-build$BUILD_NUMBER //cdn.fromdoppler.com/doppler-webapp int
sh ./build-n-publish.sh dev-build$BUILD_NUMBER //cdn.fromdoppler.com/doppler-webapp development
```

I think that it is a valid option, is more flexible, but doing that in the same _build-n-publish.sh_ call, allow us to optimize it a little bit, for example running the tests only one time.

Does it answer your question?

Reply:
Ok. thanks for the explanation. I get your point.